js/field.js

```diff
@@ -0,0 +1,324 @@
+      cardElement.classList.add('js-closed');
+      cardElement.classList.remove('js-draggable');
+    }.bind(this));
+
+    this._deck.topCardElement.onclick = this._onDeckTopCardClick.bind(this);
+  }
+
+  _addCardToHome(homeNumber, cardId) {
+    let cardElement = this._el.querySelector('[data-card-id="' + cardId + '"]');
+    let home = this['_home' + homeNumber];
+    let card = this._removeCardFromCurrentHolder(cardId);
+
+    if ((home.currentSuit === null || card.suit === home.currentSuit) &&
+      home.currentPriority === card.priority - 1) {
+
+      this._openTopCard(this._getCardCurrentHolder(cardId));
+
+      home.cards.push(card);
+      cardElement.style.zIndex = ++this._currrentZindex;
+      cardElement.style.top = home.top + 'px';
+      cardElement.style.left = home.left + 'px';
+      cardElement.setAttribute('data-current-holder-id', 'home-' + homeNumber);
+      cardElement.onclick = null;
+    } else {
+      this._returnCardToPreviousHolder(card);
+      this._dragManager.rollBack(cardElement);
+    }
+  }
+
+  _removeCardFromCurrentHolder(cardId) {
+    let currentHolderId = this._getCardCurrentHolder(cardId);
+    let holder;
+
+    if (currentHolderId.slice(0, 4) === 'home') {
+      let homeNumber = currentHolderId.slice(-1);
+      holder = this['_home' + homeNumber];
+
+      return holder.cards.pop();
+    }
+
+    if (currentHolderId.slice(0, 4) === 'pile') {
+      let pileNumber = currentHolderId.slice(-1);
+      holder = this['_pile' + pileNumber];
+
+      return holder.cards.pop();
+    }
+
+    if (currentHolderId === 'open-deck') {
+      holder = this._openDeck;
+
+      return holder.cards.pop();
+    }
+  }
+
+  _returnCardToPreviousHolder(card) {
+    let currentHolderId = this._getCardCurrentHolder(card.id);
+    let holder;
+
+    if (currentHolderId.slice(0, 4) === 'home') {
+      let homeNumber = currentHolderId.slice(-1);
+      holder = this['_home' + homeNumber];
+
+      return holder.cards.push(card);
+    }
+
+    if (currentHolderId.slice(0, 4) === 'pile') {
+      let pileNumber = currentHolderId.slice(-1);
+      holder = this['_pile' + pileNumber];
+
+      return holder.cards.push(card);
+    }
+
+    if (currentHolderId === 'open-deck') {
+      holder = this._openDeck;
+
+      return holder.cards.push(card);
+    }
+  }
+
+  _openCard(cardElement) {
+    cardElement.classList.remove('js-closed');
+    cardElement.classList.add('js-draggable');
+    cardElement.classList.add('js-droppable');
+  }
+
+  _getCardCurrentHolder(cardId) {
+    let cardElement = this._el.querySelector('[data-card-id="' + cardId + '"]');
+    return cardElement.dataset.currentHolderId;
+  }
+
+  _checkForCorrectSuit(firstSuit, secondSuit) {
+    return ((firstSuit == 'hearts' || firstSuit == 'diamonds' && secondSuit == 'clubs' || secondSuit == 'spades') ||
+            (firstSuit == 'clubs' || firstSuit == 'spades' && secondSuit == 'hearts' || secondSuit == 'diamonds'));
+  }
+}
```

Comment:
In general I like your approach with components. However it seems to get a bit messy at this point as some bugs are present and this requires investigation to solve them. Please check your code and fix bugs :)
The functions in your code are small enough and easy to understand what they are doing.